src/Http/GraphRequest.php

```diff
@@ -249,84 +190,48 @@ public function getHeaders()
     * Attach a body to the request. Will JSON encode
     * any Microsoft\Graph\Model objects as well as arrays
     *
-    * @param mixed $obj The object to include in the request
+    * @param string|StreamInterface|object $obj The object to include in the request
     *
-    * @return GraphRequest object
+    * @return $this object
     */
-    public function attachBody($obj)
+    public function attachBody($obj): self
     {
         // Attach streams & JSON automatically
-        if (is_string($obj) || is_a($obj, 'GuzzleHttp\\Psr7\\Stream')) {
+        if (is_string($obj) || is_a($obj, StreamInterface::class)) {
             $this->requestBody = $obj;
         }
         // By default, JSON-encode
         else {
             $this->requestBody = json_encode($obj);
         }
+        $this->initPsr7HttpRequest();
         return $this;
     }
 
     /**
     * Get the body of the request
     *
-    * @return mixed request body of any type
+    * @return string|StreamInterface request body
     */
     public function getBody()
     {
         return $this->requestBody;
     }
 
     /**
-    * Sets the timeout limit of the cURL request
-    *
-    * @param string $timeout The timeout in seconds
-    *
-    * @return GraphRequest object
-    */
-    public function setTimeout($timeout)
-    {
-        $this->timeout = $timeout;
-        return $this;
-    }
-
-    /**
-     * Gets the timeout value of the request
+     * Executes the HTTP request using $graphClient's http client or a PSR-18 compliant HTTP client
      *
-     * @return string
+     * @param ClientInterface|null $client (optional) When null, uses $graphClient's http client
+     * @return array|GraphResponse|StreamInterface|object Graph Response object or response body cast to $returnType
+     * @throws ClientExceptionInterface
```

Comment:
Made no sense to me to throw a `GraphClientException` here since we were only wrapping an existing standard exception interface and providing no further info.